core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigrationActor.scala

```diff
@@ -0,0 +1,857 @@
+              "billingProject" -> billingProject.projectName,
+              "billingProjectBillingAccount" -> billingProjectBillingAccount
+            )
+          )
+        }
+
+        requesterPaysEnabled <- liftIO {
+          for {
+            bucketOpt <- storageService.getBucket(
+              GoogleProject(workspace.googleProjectId.value),
+              GcsBucketName(workspace.bucketName),
+              bucketGetOptions = List(Storage.BucketGetOption.fields(Storage.BucketField.BILLING))
+            )
+
+            bucketInfo <- IO.fromOption(bucketOpt)(noWorkspaceBucketError(migration, workspace))
+            requesterPaysEnabled = Option(bucketInfo.requesterPays()).exists(_.booleanValue())
+
+            actorSaIdentity = serviceAccount(userInfo.userEmail.value)
+            _ <- storageService
+              .overrideIamPolicy(
+                GcsBucketName(workspace.bucketName),
+                Map(StorageRole.StorageAdmin -> NonEmptyList.one(actorSaIdentity)),
+                bucketSourceOptions =
+                  if (requesterPaysEnabled) List(BucketSourceOption.userProject(googleProjectToBill.value))
+                  else List.empty,
+                version = 3
+              )
+              .compile
+              .drain
+          } yield requesterPaysEnabled
+        }
+
+        now <- nowTimestamp
+        _ <- inTransaction { dataAccess =>
+          dataAccess.multiregionalBucketMigrationQuery.update2(
+            migration.id,
+            dataAccess.multiregionalBucketMigrationQuery.requesterPaysEnabledCol,
+            requesterPaysEnabled,
+            dataAccess.multiregionalBucketMigrationQuery.workspaceBucketIamRemovedCol,
+            now.some
+          )
+        }
+        _ <- getLogger[MigrateAction].info(
+          stringify(
+            "migrationId" -> migration.id,
+            "workspace" -> workspace.toWorkspaceName,
+            "workspaceBucketIamRemoved" -> now
+          )
+        )
+      } yield ()
+    }
+
+  final def createTempBucket: MigrateAction[Unit] =
+    withMigration(_.multiregionalBucketMigrationQuery.createTempBucketConditionCondition) { (migration, workspace) =>
+      for {
+        tmpBucketName <-
+          liftIO(randomSuffix("terra-multiregional-migration-").map(GcsBucketName))
+
+        _ <- createBucketInTargetRegion(
+          migration,
+          workspace,
+          sourceBucketName = GcsBucketName(workspace.bucketName),
+          destBucketName = tmpBucketName
+        )
+
+        now <- nowTimestamp
+        _ <- inTransaction { dataAccess =>
+          import dataAccess.setOptionValueObject
+          dataAccess.multiregionalBucketMigrationQuery.update2(
+            migration.id,
+            dataAccess.multiregionalBucketMigrationQuery.tmpBucketCol,
+            Some(tmpBucketName),
+            dataAccess.multiregionalBucketMigrationQuery.tmpBucketCreatedCol,
+            Some(now)
```

Comment:
Curious why this is needed given the triggers being added.

Reply:
The triggers will only update the `UPDATED` column in their tables. Other columns still need to be explicitly updated depending on what step in the pipeline has finished